exchanges/kucoin/kucoin_test.go

```diff
@@ -0,0 +1,2419 @@
+	enabledPairs, err := ku.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		t.Fatal(err)
+	}
+	startTime := time.Now().Add(-time.Hour * 5)
+	endTime := time.Now().Add(-time.Hour * 2)
+	_, err = ku.GetHistoricCandlesExtended(context.Background(), enabledPairs[0], asset.Spot, kline.OneHour, startTime, endTime)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = ku.GetHistoricCandlesExtended(context.Background(), enabledPairs[0], asset.Spot, kline.FiveMin, startTime, endTime)
+	if err != nil {
+		t.Error(err)
+	}
+	enabledPairs, err = ku.GetEnabledPairs(asset.Futures)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = ku.GetHistoricCandlesExtended(context.Background(), enabledPairs[0], asset.Futures, kline.OneHour, startTime, endTime)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = ku.GetHistoricCandlesExtended(context.Background(), enabledPairs[0], asset.Futures, kline.FiveMin, startTime, endTime)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetServerTime(t *testing.T) {
+	t.Parallel()
+	_, err := ku.GetServerTime(context.Background(), asset.Spot)
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = ku.GetServerTime(context.Background(), asset.Futures)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetRecentTrades(t *testing.T) {
+	t.Parallel()
+	enabledPairs, err := ku.GetEnabledPairs(asset.Futures)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = ku.GetRecentTrades(context.Background(), enabledPairs[0], asset.Futures)
+	if err != nil {
+		t.Error(err)
+	}
+	enabledPairs, err = ku.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = ku.GetRecentTrades(context.Background(), enabledPairs[0], asset.Spot)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetOrderHistory(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+	var getOrdersRequest = order.GetOrdersRequest{
+		Type:      order.Limit,
+		Pairs:     []currency.Pair{currency.NewPair(currency.LTC, currency.BTC), currency.NewPair(currency.MHC, currency.ETH), currency.NewPair(currency.MHC, currency.BTC), currency.NewPair(currency.OXEN, currency.BTC)},
+		AssetType: asset.Futures,
+		Side:      order.Sell,
+	}
+	_, err := ku.GetOrderHistory(context.Background(), &getOrdersRequest)
+	if err != nil {
+		t.Error(err)
+	}
+	getOrdersRequest.Pairs = []currency.Pair{}
+	_, err = ku.GetOrderHistory(context.Background(), &getOrdersRequest)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestGetActiveOrders(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credentialsNotSet)
+	}
+	enabledPairs, err := ku.GetEnabledPairs(asset.Spot)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var getOrdersRequest = order.GetOrdersRequest{
+		Type:      order.Limit,
+		Pairs:     enabledPairs,
+		AssetType: asset.Futures,
+		Side:      order.Buy,
+	}
+	if _, err = ku.GetActiveOrders(context.Background(), &getOrdersRequest); err != nil {
+		t.Error("Kucoin GetActiveOrders() error", err)
+	}
+	getOrdersRequest.Pairs = []currency.Pair{}
```

Comment:
This scenario cannot work as you send enabled SPOT pairs for a futures request. Could you seperate these out?